feat: expose async wrap providers under async_hooks module (#40760)
nodejs/node: doc/api/async_hooks.md

Promise contexts may not get valid `triggerAsyncId`s by default. See
the section on [promise execution tracking][].

### `async_hooks.asyncWrapProviders`

<!-- YAML

> added: v?.?.?

Author: How I should fill it?

Reviewer: TODO

Reviewer 2: ```suggestion
added: REPLACEME
```

> added: v?.?.?

Reviewer 3: ```suggestion
added: REPLACEME
```

-->

> * Returns: A list of providers supported by async wrap.

Reviewer 3: Do we define anywhere what a "provider" is?
Also, to me "A list" means an array, but an object is returned.

Author: I think that it wasn't defined, at least publically. I can reword that for sure, but, to be honest, I don't know how to define the `providers` description.

Reviewer 4: ```suggestion
* Returns: A map of provider types supported by async wrap.
```

I would also add something to the description linking it to `type` parameter of the `init(...)` hook.

Author: Just did, thanks for point that out!

Promise contexts may not get valid `triggerAsyncId`s by default. See
the section on [promise execution tracking][].

### `async_hooks.asyncWrapProviders`

<!-- YAML
added: REPLACEME
-->

* Returns: A map of provider types to the corresponding numeric id.

>   Those map contain all the events that might be emitted by `async_hooks.init()` event.

Reviewer 4: ```suggestion
  This map contains all the event types that might be emitted by the `async_hooks.init()` event.
```

> This feature suppress the deprecated usage of `process.binding('async_wrap').Providers`.

Reviewer 5: ```suggestion
This feature suppresses the deprecated usage of `process.binding('async_wrap').Providers`.
```

> This feature suppresses the deprecated usage of `process.binding('async_wrap').Providers`.

Reviewer 6: Maybe add a link to DEP0111
